File: Events/src/main/java/com/inverce/mod/events/ChannelGroup.java
package com.inverce.mod.events;

import android.util.SparseArray;

public class ChannelGroup {
    protected final boolean useWeekEvents;
    protected SparseArray<Channel> channels;

    public ChannelGroup() {
        this(true);
    }

    public ChannelGroup(boolean useWeekEvents) {
        this.useWeekEvents = useWeekEvents;
        this.channels = new SparseArray<>();
    }

    public Channel on(int channelId) {
        Channel channel = channels.get(channelId);
        if (channel != null) {
            return channel;
        }
        channels.put(channelId, channel = new Channel(useWeekEvents));
        return channel;
    }

}

File: Events/src/main/java/com/inverce/mod/v2/events/ChannelGroup.kt
package com.inverce.mod.v2.events

import android.util.SparseArray

internal interface IChannelGroup {
    fun on(channelId: Int): Channel
}

open class ChannelGroup(protected val useWeekEvents: Boolean = true): IChannelGroup {
    protected var channels = SparseArray<Channel>()

    override fun on(channelId: Int): Channel {
        return channels.get(channelId) ?: Channel(useWeekEvents).apply {
            channels.put(channelId, this)
        }
    }
}